syntax = "proto3";

package microservices;

// Task message sent from ASP.NET to Python worker
message TaskMessage {
  string task_id = 1; // Unique identifier for the task
  string task_type = 2; // Type of task to be performed
  string data = 3; // Data required for the task
  int64 timestamp = 4; // Timestamp when the task was created
}

// Result message sent from Python worker to ASP.NET
message ResultMessage {
  string task_id = 1;
  string status = 2;
  string result = 3;
  int64 timestamp = 4;
}

message QuerySearchRequest {
  string query = 1;
  int32 max_results = 2;
}

message QuerySearchResult{
  string title = 1;
  string url = 2;
}

message AlbumInfo {
  int32 id = 1;
  string title = 2;
  string artist = 3;
  int32 year = 4;
  repeated string tracks = 5;
}

message ImageInfo {
  int32 album_id = 1;
  bytes image_data = 2;
}
